src/System.CommandLine/ParseResultExtensions.cs

```diff
@@ -9,6 +9,36 @@ namespace System.CommandLine
 {
     public static class ParseResultExtensions
     {
+        public static object GetDefaultValue(this Type type)
+        {
+            return type.IsValueType ? Activator.CreateInstance(type) : null;
+        }
+
+        public static object GetValue(this ParseResult parseResult, Option option)
+        {
+            var result = parseResult.CommandResult.Children
+                                   .Where(c => c.Symbol == option)
+                                   .FirstOrDefault();
+            switch (result)
+            {
+                case OptionResult optionResult:
+                    return optionResult.GetValueOrDefault();
+                case null:
+                    var optionType = option.Argument.ArgumentType == null
+                                    ? typeof(bool)
+                                    : option.Argument.ArgumentType;
+                    return optionType.GetDefaultValue();
+                default:
+                    throw new InvalidOperationException("Internal: Unknown result type");
+            }
+        }
+
+        public static object GetValue(this ParseResult parseResult, Argument argument)
```

Comment:
I believe this should be called `GetValueOrDefault`. The naming is meant to follow familiar LINQ method conventions, e.g.:

* `Enumerable.FirstOrDefault` -> Returns the first item or the type default. 
* `Enumerable.First` -> Returns the first item or throws.